_includes/manuals/1.24/5.7.7_openid_sso.md

```diff
@@ -0,0 +1,106 @@
+
+Procedure
+
+1. Install the mod_auth packages:
+```
+# yum install mod_auth_openidc python2-requests-oauthlib keycloak-httpd-client-install -y
+```
+
+2. Register Katello/Foreman as application in the Keycloak:
+```
+# export KEYCLOAK_URL=https://keycloak.example.com
+# export KEYCLOAK_REALM=ssl-realm
+# export KEYCLOAK_USER=admin
+# keycloak-httpd-client-install --app-name foreman-openidc --keycloak-server-url $KEYCLOAK_URL --keycloak-admin-username $KEYCLOAK_USER --keycloak-realm $KEYCLOAK_REALM --keycloak-admin-realm master --keycloak-auth-role root-admin -t openidc -l /users/extlogin -d
+```
+Warning: This feature is not yet supported by foreman-installer. As a result, re-running the foreman-installer command can delete the changes in Apache files added by this feature.
+
+3. Log in to the Foreman admin page and click **Administer** → **Settings** → **Authentication**.
+ ![Foreman UI](Foreman_UI.png)
+
+4. Set **authorize_login_delegation_auth_source_user_autocreate** to **External** to enable auto-creation of users from external OpenID provider.
+
+5. Set the following OIDC parameters:
+
+ **OIDC Algorithm**: Algorithm type with which JWT was encoded in the OpenID provider. For example, For example, set this parameter to RS-256.
+
+ **OIDC Audience**: Name of the OIDC audience used for authentication. In case of [Keycloak](https://www.keycloak.org/), it is Client ID.
+
+ **OIDC Issuer**: The Issuer (iss) claim identifies the principal that issues the JWT.
+
+ **OIDC JWKS URL**: Open JSON Web key Set URL to validate the signature.
+
+ ![Foreman UI](Foreman_UI_Settings.png)
+
+#### End-user
+
+An end user can perform SSO by the following ways:
+* User Interface
+* Hammer-CLI
+
+To perform the mentioned ways, ensure that Foreman is configured. For more information, see Configuring Foreman.
+
+##### User Interface
+
+Procedure
+
+* Log in to the https://foreman.example.com/users/extlogin
+
+ If credentials are correct, it redirects to foreman dashboard.
+
+##### Hammer-cli
+
+Hammer-cli supports the following methods to obtain ID token and perform authentication:
+
+* Authorization Code Grant Flow
+* Password Grant Flow
+
+###### Authorization Code Grant Flow
+
+Authorization Code Flow is a two step process:
+
+* Authorization Code Flow returns an Authorization Code to the client.
+* this Code can be exchanged for an access token directly.
+
+Procedure
+
+1. Get the token endpoint and authorization endpoint from the .well-known/openid-configuration URL of your OpenID provider. For example in case of Keycloak, get token and authorization endpoint from: https://<keycloak.example.com>/auth/realms/<realm-name>/.well-known/openid-configuration
+
+2. Log in to Foreman:
+```
+hammer auth login oauth --two-factor --oidc-token-endpoint https://keycloak.exmaple.com/token --oidc-authorization-endpoint https://keycloak.example.com/auth --oidc-client-id foreman-client --oidc-redirect-uri urn:ietf:wg:oauth:2.0:oob
```

Comment:
Does `--oidc-client-id foreman-client` in this line correspond to `keycloak-httpd-client-install --app-name foreman-openidc` from line 35? If they're same, perhaps use the same names to avoid confusion. If not, please ignore.